frontend/src/apps/qotd/containers/Result.js

```diff
@@ -10,25 +10,24 @@ import Unavailable from '../components/Unavailable'
 
 import { getQotd } from '../actions'
 
-class Answer extends React.Component {
-  state = { fetchFailed: false }
+function Answer(props) {
+  const [fetchFailed, setfetchFailed] = useState(false)
+  const { qotd } = props
 
-  componentDidMount () {
-    const { qotd } = this.props
+  useEffect(() => {
     // Only fetch qotd if not already in redux or more than 24 hours old
     if (!qotd || Date.now() - new Date(qotd.day) > 24 * 60 * 60 * 1000) {
-      this.props.getQotd().catch(() => this.setState({ fetchFailed: true }))
+      props.getQotd().catch(() => setfetchFailed(true))
     }
-  }
+  }, [qotd, props])
```

Comment:
Having `props` in the dependency array means that the `useEffect` callback runs every time the `props` object is modified. We only care about `qotd` and `getQotd` so the dependency array should be `[qotd, getQotd]`. Make sure to destructure `getQotd` outside of the `useEffect` callback.